Sources/POSIX/Error.swift

```diff
@@ -36,14 +36,26 @@ import func libc.strerror
 
 
 extension SystemError: CustomStringConvertible {
-    public var description: String {
-
+    public var description: {
         func strerror(_ errno: Int32) -> String {
-            let cmsg = libc.strerror(errno)!
-            let msg = String(validatingUTF8: cmsg) ?? "Unknown Error"
-            return "\(msg) (\(errno))"
+            var cap = 64
+            while true {
+                var buf = [Int8](repeating: 0, count: cap)
+                let err = strerror_r(errno, &buf, buf.count)
+                if err == EINVAL {
+                    return "Unknown error \(errno)"
+                }
+                if err == ERANGE {
+                    cap *= 2
```

Comment:
If a syscall misbehaves, we are in an undefined state anyway. Wouldn't it be better to crash cleanly in that scenario than trying to save this part with heuristics?